### PR TITLE
📢 Add submission publish/unpublish to CLI

### PR DESCRIPTION
**Pull request by fwkoch**
Now after `curvenote submit [venue]`, you may `curvenote submission publish [venue]` and then if you made a mistake `curvenote submission unpublish [venue]` 😳

### CODE REVIEW

File: packages/curvenote-cli/src/submissions/submit.ts [hunk 1]
```diff
@@ -214,9 +217,16 @@ export async function submit(session: ISession, venue: string, opts?: SubmitOpts
...
     });
 
-    // const cdnKey = 'ad7fa60f-5460-4bf9-96ea-59be87944e41'; // dev debug
     const cdn = opts?.draft ? session.TEMP_CDN : session.PRIVATE_CDN;
-    const { cdnKey } = await uploads.uploadToCdn(session, cdn, opts);
+    let cdnKey: string;
+    if (!process.env.DEV_CDN || process.env.DEV_CDN === 'false') {
+      const uploadResult = await uploads.uploadToCdn(session, cdn, opts);
+      cdnKey = uploadResult.cdnKey;
+    } else if (process.env.DEV_CDN.match(CDN_KEY_RE)) {
```
Comment (fwkoch) [inline packages/curvenote-cli/src/submissions/submit.ts:225]: This makes the previously commented-out `dev debug` more explicit.

If you do not want to hit the cdn, you may use default content with `DEV_CDN=true curvenote submit [venue]`

Or you can specify other existing content with `DEV_CDN=my-existing-cdn-uuid curvenote submit [venue]`

File: packages/curvenote-cli/src/submissions/submit.utils.ts [hunk 2]
```diff
@@ -259,12 +259,12 @@ export async function confirmUpdateToExistingSubmission(
     process.exit(1);
   }
 
-  if (opts?.kind) {
+  const kind = existingSubmission?.kind;
+  if (opts?.kind && opts.kind !== kind) {
```
Comment (fwkoch) [inline packages/curvenote-cli/src/submissions/submit.utils.ts:263]: This just eliminates the note about ignored `kind` if you are simply re-specifying the correct kind.

File: packages/curvenote-cli/src/submissions/submit.ts [hunk 3]
```diff
@@ -26,6 +26,9 @@ import { getChecksForSubmission } from './check.js';
 import { getGitRepoInfo } from './utils.git.js';
 import * as uploads from '../uploads/index.js';
 
+const CDN_KEY_RE = /^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$/;
```
Comment (stevejpurves) [inline packages/curvenote-cli/src/submissions/submit.ts:29]: basically this is effectvily a `isUUID()` check? thats function doesn't exist or anything, just suggesting that this is maybe more broadly useful than the cdn key

**Review by stevejpurves — COMMENTED** [review]
looks great! might need to change based on comments on the other PR in `journals`

**Pull request merged**